SalesforceCommerceCloud/pwa-kit
# [Spike] Extending analytics with template extensibility - @W-12734379@

This spike explores ways on extending the current Einstein analytics:
- adding another analytics provider (like Google Analytics)
- adding more events (like custom events that are unique to your projects)

By going through the above scenarios, it will give us feedback on our template extensibility too.

## How to test the PR

1. `cd packages/my-extended-retail-app` and `npm start`
2. Open up your browser console
3. Verify that the log `--- can send pageview event to other analytics providers` shows up. Note that in this PR, I used console logs in a few places to simulate sending events.
4. Verify in the Network tab that filtering "google" shows activities from Google Analytics
5. Scroll down the page to the footer section
6. Click on the Sign Up button
7. Verify that there are new console logs simulating the sending of custom events
8. Type `window.dataLayer` into the console and verify that inside GA's dataLayer array, its last item is the `email_signup` custom event.
